### ISSUE TITLE
How to connect two PC using Ethernet Cable?

### ISSUE THREAD

**Issue by AKSKMY** [debauchee/barrier#930]
By default what I did was to connect both laptop and PC using one Ethernet Cable and then change both IP address to be within the same gateway but different IP addresses. But still doesn't seem to work. I'm wanting to shift to LAN cuz Wifi isn't stable at all for me. Once high ping spike the keyboard and mouse sharing becomes totally unusable. Pls advice, thanks!

**Comment by MountNoperest** [comment]
I got it working with Ubuntu 18.04 and the latest version of Manjaro kernel:'5.4.72-1-MANJARO.

 This is how I did it. 

1. I connected the ethernet cable to each computers ethernet port. 
2. I went into the system settings "edit connection" and configured the "wired connection 1"
![wired](https://user-images.githubusercontent.com/65341112/97494782-15066a80-1967-11eb-84d4-4c753e72dacc.png)
3. In the configuration of the wired connection, go to IP4-settings. 
4. In the "method dropdown menu choose "Only local link" and check off the "require IP4 addresses to complete the connection"
![connections](https://user-images.githubusercontent.com/65341112/97495093-8b0ad180-1967-11eb-8fcd-e389186da5ed.png)
5. Do this for the other computer as well. 
6. Setup Barrier as usual. 

I hope this works for you.

**Comment by roflcoopter** [comment]
Any idea how to do this on windows?

**Comment by MountNoperest** [comment]
@roflcoopter Yes somewhat. My windows computer assigned an automatic IP adress to the ethernet cable when it was connected to the Manjaro computer. The connection took a while but after it assigned the IP adress, I restarted barrier and connected to the ethernet IP adress. The Manjaro computer was configured the same way as mentioned in my other comment. 
I don't know how to connect windows to windows computer, since I only own 1 windows pc. 

I hope this helps.

**Comment by 91imp91** [comment]
You made mistake with setting your lan network like this. If your wifi connection is not stable, then you only need to connect both your pc's to home router via cable. If you connect 2 pc's together you lose your connection to Internet. Anyway, there what you need to do if you still want get things your way:
1: set ip address of server pc to 192.168.0.2 (subnet mask 255.255.255.0, gateway empty(not zeros))
2: set ip address of client pc to 192.168.0.3 (subnet mask 255.255.255.0, gateway 192.168.0.2)
3: set barrier settings acording to those settings, and that it.

**Comment by MountNoperest** [comment]
A thank you @91imp91 for the clarification. When I tested with the Ethernet cable directly connected to each PC through the Ethernet port, I didn't experience any loss of Wi-Fi on either Linux to Linux, or Linux to Windows connected and configured like this. I don't know why, maybe my Wi-Fi is more stable than I thought!?

**Comment by namefailed** [comment]
Is there any benefit to doing this? I am unable to connect directly to the router in my current situation, but I can connect the Client & Server together via an ethernet cable. Would this reduce some of the lag/hiccups caused by my, sometimes amazing, sometimes garbo, internet connection? 

Thanks!

**Comment by AKSKMY** [comment]
Btw I end up using a router (which provided internet as well) to host my barrier connection as a bridge between two PCs. Works fine most of the time.

**Comment by nbolton** [comment]
It might be helpful to others to continue this discussion [upstream](https://github.com/deskflow/deskflow) in Deskflow, as Barrier is no longer in development. The Input Leap (fork) community is a bit more active than this one, but not as active as upstream Deskflow.

**Comment by AKSKMY** [comment]
> It might be helpful to others to continue this discussion [upstream](https://github.com/deskflow/deskflow) in Deskflow, as Barrier is no longer in development. The Input Leap (fork) community is a bit more active than this one, but not as active as upstream Deskflow.

Hi I see! Just wondering is Deskflow an improved version of Barrier? I'm still using Barrier as back then there wasn't anything else like Barrier.

Update** I shift to Deskflow but seems like the TLS encryption has issues had to turn it off. Other than that can't see much ground breaking difference haha. Will see how it goes!

Haha saw your website @nbolton https://symless.com/synergy/news/why-did-barrier-fail cool!

**Comment by nbolton** [comment]
> Just wondering is Deskflow an improved version of Barrier? 

Deskflow is the upstream of Barrier; that is to say Barrier is a fork of Deskflow. It was forked around v1.9(?) of Deskflow (which is now at v1.24)... so quite a long time ago. There are some improvements made in Barrier that have [not yet been ported](https://github.com/deskflow/deskflow/wiki/Backport) back upstream to Deskflow. So, in some ways it's better, and in some ways still has some aspects to improve. If you're wondering about Synergy, that is now downstream of Deskflow.

> Update** I shift to Deskflow but seems like the TLS encryption has issues had to turn it off. 

What issues did you have with TLS? Perhaps you can link to the issue on Deskflow?

> Haha saw your website [@nbolton](https://github.com/nbolton) https://symless.com/synergy/news/why-did-barrier-fail cool!

Just trying to help everyone understand what's going on! :)
